src/scm/Model.ts

```diff
@@ -150,39 +151,122 @@ export class Model implements Disposable {
         } else if (id.startsWith('pending')) {
             const command = 'describe';
             const args = id.substr(id.indexOf(':') + 1);
-            const uri: Uri = new Uri().with({scheme: 'perforce', authority: command, query: args});
+            const uri: Uri = new Uri().with({ scheme: 'perforce', authority: command, query: args });
             commands.executeCommand<void>("vscode.open", uri);
         }
     }
 
+    public async SubmitDefault(): Promise<void> {
+        const loggedin = await Utils.isLoggedIn(this._compatibilityMode);
+        if (!loggedin) {
+            return;
+        }
+
+        const noFiles = 'File(s) not opened on this client.';
+        let fileListStr;
+        try {
+            fileListStr = await Utils.getOutput('opened', null, null, '-c default');
+            if (fileListStr === noFiles) {
+                Display.showError(noFiles);
+                return;
+            }
+        } catch (err) {
+            Display.showError(err.toString());
+        }
+
+
+        const fileList = fileListStr.split("\n").map(file => {
+            const endOfFileStr = file.indexOf('#');
+            return file.substring(0, endOfFileStr);
+        });
+
+        const descStr = await vscode.window.showInputBox({
+            placeHolder: 'New changelist description',
+            validateInput: (value: string) => {
+                if (!value || value.length === 0) {
+                    return 'Cannot set empty description';
+                }
+                return null;
+            },
+            ignoreFocusOut: true
+        });
+
+        if (descStr === undefined) {
+            return;
+        }
+
+        const pick = await vscode.window.showQuickPick(["Submit", "Save Changelist", "Cancel"], { ignoreFocusOut: true });
+
+        let command;
+        let args = '';
+
+        if (!pick || pick == "Cancel") {
+            return;
+        }
+
+        if(pick === "Submit") {
+            this.Submit(descStr);
+            return;
+        }
+
+        // Save to changelist
```

Comment:
Same part of code as `ProcessChangelist()`
Should be moved into its own private function (todo for later)